install.sh

```diff
@@ -22,86 +92,304 @@ deps=(
+    for i in ${dependencies[@]}; do
+      installString+=" $i"
+    done
+
+    #run install
+    echo ${installString}
+    ${installString}
+    if [[ $? -eq 0 ]]; then
+        echo -e All dependencies Installed ok '\n'
+    else
+        echo Package failed to install with error code $?, quitting check logs above
+        exit
+    fi
 fi
 
-#begin cmake
-echo "beginning cmake for raspberry pi"
-cmake -DRPI_BUILD=TRUE -DGST_BUILD=TRUE -DCMAKE_BUILD_TYPE=Release . 
-if [[ $? > 0 ]]
-  then
-    echo "Cmake error"
-  exit
+
+###############################  AASDK #########################
+if [ $aasdk = false ]; then
+	echo -e Skipping aasdk '\n'
 else
-  echo "Cmake ok"
-  echo #####
-fi
+  #change to parent directory
+  cd ..
 
-#start make
-echo "beginning make, this will take a while"
-make
-if [[ $? > 0 ]]
-  then
-    echo "make error check output above"
+  #clone aasdk
+  git clone $aasdkRepo
+  if [[ $? -eq 0 ]]; then
+    echo -e Aasdk Cloned ok '\n'
+  else
+    cd aasdk
+    if [[ $? -eq 0 ]]; then
+      git pull $aasdkRepo
+      echo -e Aasdk Cloned OK '\n'
+      cd ..
+    else
+      echo Aasdk clone/pull error
+      exit
+    fi
+  fi
+
+  #change into aasdk folder
+  echo -e moving to aasdk '\n'
+  cd aasdk
+
+  #beginning cmake
+  cmake -DCMAKE_BUILD_TYPE=Release .
+  if [[ $? -eq 0 ]]; then
+      echo -e Aasdk CMake completed successfully'\n'
+  else
+    echo Aasdk CMake failed with code $?
     exit
+  fi
+
+  #beginning make
+  make -j2
+
+  if [[ $? -eq 0 ]]; then
+    echo -e Aasdk Make completed successfully '\n'
   else
-    echo "make ok, executable can be found ../bin/ia"
+    echo Aasdk Make failed with code $?
+    exit
+  fi
+
+  #begin make install
+  sudo make install
+
+  if [[ $? -eq 0 ]]
+    then
+    echo -e Aasdk installed ok'\n'
     echo
+  else
+    echo Aasdk install failed with code $?
+    exit
+  fi
+  cd ../dash
+fi
 
-    #check if usb rules exist
-    echo "checking if permissions exist"
 
-    #udev rule to be created below, change as needed
-    FILE=/etc/udev/rules.d/51-iadash.rules
-    if [[ -f "$FILE" ]]
-      then
-        echo "rules exists"
-        echo
-      else
-        sudo touch $FILE
+###############################  gstreamer  #########################
+#check if gstreamer install is requested
+if [ $gstreamer = true ]; then
+  echo installing gstreamer
 
-        # OPEN USB RULE, CREATE MORE SECURE RULE IF REQUIRED
-        echo "SUBSYSTEM==\"usb\", ATTR{idVendor}==\"*\", ATTR{idProduct}==\"*\", MODE=\"0660\", GROUP=\"plugdev\"" | sudo tee $FILE
-      if [[ $? > 0 ]]
-        then
-          echo "unable to create permissions"
-        else
-          echo "permissions created"
+  #change to parent directory
+  cd ..
+
+  #clone gstreamer
+  echo Cloning Gstreamer
+  git clone $gstreamerRepo
+  if [[ $? -eq 0 ]]; then
+    echo -e Gstreamer cloned OK
+  else
+    cd qt-gstreamer
+      if [[ $? -eq 0 ]]; then
+        git pull gstreamerRepo
```

Comment:
same here:  Please add check if command was successful, as script continues in case of error:

Reply:
I'd guess it's debatable whether you would want it to stop, if you are working and have changes, you would want it to build with the changes rather then exit on commits not added?